core/textile/client.go

```diff
@@ -172,6 +178,52 @@ func (tc *textileClient) SetUc(uc UsersClient) {
 	tc.uc = uc
 }
 
+func (tc *textileClient) SetMb(mb Mailbox) {
+	tc.mb = mb
+}
+
+func (tc *textileClient) setupMailBox(ctx context.Context) error {
+	maillib := mail.NewMail(cmd.NewClients("api.textile.io:443", true), mail.DefaultConfConfig())
+
+	usr, _ := user.Current()
+	mbpath := filepath.Join(usr.HomeDir, ".fleek-space/textile/mail")
+
+	var mailbox *mail.Mailbox
+	dbid, err := tc.store.Get([]byte(mailboxSetupFlagStoreKey))
+	if err == nil && len(dbid) > 0 {
+		// restore
+		mailbox, err = maillib.GetLocalMailbox(ctx, mbpath)
+		if err != nil {
+			return err
+		}
```

Comment:
Maybe `return mailbox, nil`  here to avoid the `else`.

Or you can wrap this into a func like `GetMailboxWithCache`.

Also wondering if some of this init logic can be moved to `mailbox.go`.

Reply:
Fixed in [5a39569](https://github.com/FleekHQ/space-daemon/pull/134/commits/5a3956938857a5752d199d3ea9223ca4ca983837). Note that the `else` was one level above so I wasn't sure how to remove it.  However I moved the mailbox init into mailbox.go and refactored into 2 smaller functions.